aten/src/ATen/NestedTensorImpl.h

```diff
@@ -63,6 +66,7 @@ struct TORCH_API NestedTensorImpl : public c10::TensorImpl {
 
   at::Tensor buffer_;
   const at::Tensor nested_size_tensor_;
+  std::vector<c10::optional<int64_t>> opt_sizes_;
```

Comment:
this is not a memory-efficient data structure if any integer value could be used to indicate unoccupied; for example, is -1 a valid value? if not, you could store with -1 internally and convert to c10::optional<int64_t> when returning from get_opt_size